### ISSUE TITLE
O7S Timeline

### ISSUE THREAD

**Issue by xephero** [quisquous/cactbot#36]
Tested for the first half or so of the fight. It basically looks for Shockwave after Load to decide which version of the timeline to run, and the latter half is untested just cause it's how far my group is. Hopefully it helps!

[o7s.txt](https://github.com/quisquous/cactbot/files/1692442/o7s.txt)

**Comment by xephero** [comment]
It looks like we both took the same approach for using +1000 for the split, just called it different things. However, I've since learned that biblio/dadaluma can be in either order in either version of the fight, so that will take some more log crawling and jumping and stuff.

**Comment by xDemonessx** [comment]
I tested @xephero on a first attempt yesterday, and it was incredibly helpful. We got up to AirForce / Bomb detonation and from what I was able to tell, the script was diverging properly every time. 

[in case you didn't see the infographic timeilne](https://i.imgur.com/FphDCUj.png)

**Comment by xephero** [comment]
I'm glad it was useful!

I've done some investigation (including that graphic) and revised my timeline. First, as I saw mentioned in a few reddit threads, there's a second split after the first one. Dadaluma and Bibliotaph will always be the second and third programs following virus, but the order of them can be either way on either route. It also seems that Load/Skip/Retrieve is sort of randomized as well, but that doesn't matter as much.

[o7s.txt](https://github.com/quisquous/cactbot/files/1697703/o7s.txt)

So, since it got kind of hard to follow, I've attached a few other things besides my timeline. The notes file is fflogs links and timestamps/casts I made when I was trying to discover a pattern between load/skip/retrieve and the actual program result. There wasn't one, but it does serve as evidence of the second split if you're looking for it. Second, I've attached diagrams; one shows the fight how it actually happens time-wise, and the other shows how they're laid out linearly in the timeline file.

[o7s-timeline-notes.txt](https://github.com/quisquous/cactbot/files/1697706/o7s-timeline-notes.txt)
[o7s-flowchart](https://user-images.githubusercontent.com/2997496/35841354-5d3b2412-0ab0-11e8-96d8-77f1608b63c0.png)
[o7s-timeline-order](https://user-images.githubusercontent.com/2997496/35841365-6c09b800-0ab0-11e8-9e92-df639fdddccc.png)

For the second split I just used additional thousands, I figure it can work as a namespace of sorts.

As for specific timeline changes:
- Added the aforementioned second split, naming them A1/A2 and B1/B2 routes. Organizationally, A and 1 mean Bibliotaph first, B and 2 mean Dadaluma first.
- Added Radars, based on your static's timeline. Weirdly, I can't find any cast or buff associated with this at all, it happens totally stealth.
- Added boss jumps for targetability, per your static's timeline. There are abilities for this, 2937 is takeoff and 2938 is landing.
- Changed Missile to Prey based on your static's timeline and a friend's suggestion, to avoid confusion with Missile Simulation. I generally prefer to stick to in-game names just to learn them and to make talking about the fight easier, but in this case it has the opposite effect, and Prey is much clearer.
- Changed Air Force's Diffractive Laser to Plane Laser for the same reason
- Changed the sync jump to be tied to gaining the Dadaluma/Bibliotaph Simulation buffs rather than the Shockwave cast. Your static's timeline was better in this regard; the buff happens 1 second after Load, and 2 seconds before the actual simulation (demon/shockwave) starts. I believe that's the earliest determination in the log that reveals the split, since whatever's on the monitor doesn't seem to make a log line and everything leading up to that seems identical.

**Comment by quisquous** [comment]
Ooh.  This is awesome.  Yeah, I was going to suggest Radars and Missile =>Prey as well.

Yeah, we saw the second path tonight and I was pretty surprised to see it.  The timeline was :100: up to that point though. https://www.twitch.tv/videos/226202843?t=8m36s was the divergence point for us from your previous script.

Are your jumps on the B path correct at time 1481? I would have expected those to jump to ~4480 and ~548, but maybe I'm misreading.

**Comment by xephero** [comment]
You're totally right, just an oversight from when I copied those lines in the A route. They're 4481 and 5481 (since he gets the buff 1 second after load/skip). Here's the corrected version:

[o7s.txt](https://github.com/quisquous/cactbot/files/1697959/o7s.txt)

And thanks for the video, I'm pretty glad it was accurate through the stuff I haven't even seen myself yet! Hopefully Aether Rot and Radar being on there will make it even nicer.

**Comment by quisquous** [comment]
Updated o7s timeline worked like 90%.  There's a few typos (missing quotes, extra colons), but I've got those fixed locally, and I'll commit this.
